graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +39,340 @@ def __init__(self, query_ast):
+
+    Returns:
+        Field object, with field_name as its name, containing directives from any field in the
+        input selections with the same name and directives from the input list of directives
+    """
+    new_field = Field(
+        name=Name(value=field_name),
+        directives=[],
+    )
+
+    # Check parent_selection for existing field of given name
+    parent_field = try_get_ast_by_name_and_type(selections, field_name, Field)
+    if parent_field is not None:
+        # Existing field, add all its directives
+        directives_from_existing_field = parent_field.directives
+        if directives_from_existing_field is not None:
+            new_field.directives.extend(directives_from_existing_field)
+
+    # Transfer directives from edge
+    if directives_from_edge is not None:
+        for directive in directives_from_edge:
+            if directive.name.value == OutputDirective.name:  # output illegal on vertex field
+                raise GraphQLValidationError(
+                    u'Directive "{}" is not allowed on a vertex field, as @output directives '
+                    u'can only exist on property fields.'.format(directive)
+                )
+            elif directive.name.value == OptionalDirective.name:
+                if try_get_ast_by_name_and_type(
+                    new_field.directives, OptionalDirective.name, Directive
+                ) is None:
+                    # New optional directive
+                    new_field.directives.append(directive)
+            elif directive.name.value == FilterDirective.name:
+                new_field.directives.append(directive)
+            else:
+                raise AssertionError(
+                    u'Unreachable code reached. Directive "{}" is of an unsupported type, and '
+                    u'was not caught in a prior validation step.'.format(directive)
+                )
+
+    return new_field
+
+
+def _get_child_type_and_selections(ast, type_info):
+    """Get the type (root field) and selection set of the input AST.
+
+    For instance, if the root of the AST is a vertex field that goes to type Animal, the
+    returned type name will be Animal.
+
+    If the input AST is a type coercion, the root field name will be the coerced type rather
+    than the union or interface type, and the selection set will contain actual fields
+    rather than a single inline fragment.
+
+    Args:
+        ast: type Node, the AST that, if split off into its own Document, would have the the
+             type (named the same as the root vertex field) and root selections that this
+             function outputs
+        type_info: TypeInfo, used to check for the type that fields lead to
+
+    Returns:
+        Tuple[str, List[Union[Field, InlineFragment]]], name and selections of the root
+        vertex field of the input AST if it were made into its own separate Document
+    """
+    child_type = type_info.get_type()  # GraphQLType
```

Comment:
It doesn't seem like `type_info` is actually used in this function outside of this line. Perhaps we can just send in the `child_type` value directly, and not make this function worry about how we actually know the type?